src/item.h

```diff
@@ -107,6 +108,8 @@ class item_category
 
 class item : public JsonSerializer, public JsonDeserializer, public visitable<item>
 {
+    friend item_location;
```

Comment:
Item should not be a friend of item_location.  Item is a critically central class and it's encapsulation shouldn't be violated.
Frankly outside of some weird situations where one class "owns" another, friend shouldn't be used.